lib/xcodeproj/project/object/native_target.rb

```diff
@@ -416,6 +416,19 @@ class PBXNativeTarget < AbstractTarget
         # @!group Helpers
         #--------------------------------------#
 
+        # Checks product_type to determine whether or not the recevier is a
+        # an app extension
+        #
+        # @return True if the target would be an extension embedded in an
+        #         app target, otherwise false
+        #
+        # @note watchOS 2 extensions are extensions of a watch target, not
+        #       an app target, whereas this is not the case for watchOS 1
+        def app_extension?
```

Comment:
That naming (and maybe this method) is more targeted toward how it's going to be used in CocoaPods. Maybe this should just be a method/helper on `AggregateTarget` in CocoaPods that checks the `user_target`'s `product_type`? The other methods in this PR could be refactored to use the existing helper `extension_target_type?` Thoughts?